src/gsym/resolver.rs

```diff
@@ -93,6 +100,88 @@ impl SymResolver for GsymResolver {
         None
     }
 
+    /// Finds the source code location for a given address.
+    ///
+    /// This function takes in an address and returns the file path,
+    /// line number and column of the line in the source code that
+    /// the address corresponds to. If it doesn't find any match it
+    /// returns `None`.
+    ///
+    /// # Arguments
+    ///
+    /// * `addr` - The address to find the source code location for.
+    ///
+    /// # Returns
+    ///
+    /// The `AddressLineInfo` corresponding to the address or `None`.
+    fn find_line_info(&self, addr: u64) -> Option<AddressLineInfo> {
+        let addr = addr.checked_sub(self.loaded_address)?;
+        let idx = find_address(&self.ctx, addr)?;
+        let symaddr = self.ctx.addr_at(idx)?;
+        if addr < symaddr {
+            return None;
+        }
+        let addrinfo = self.ctx.addr_info(idx)?;
+        if addr >= (symaddr + addrinfo.size as u64) {
+            return None;
+        }
+
+        let addrdatas = parse_address_data(addrinfo.data);
+        for adr_ent in addrdatas {
+            if adr_ent.typ != InfoTypeLineTableInfo {
+                continue;
+            }
+            // Continue to execute all GSYM line table operations
+            // until the end of the buffer is reached or a row
+            // containing addr is located.
+            let (lntab_hdr, hdr_bytes) = parse_line_table_header(adr_ent.data)?;
+            let ops = &adr_ent.data[hdr_bytes..];
+            let mut lntab_row = LineTableRow::line_table_row_from(&lntab_hdr, symaddr);
+            let mut last_lntab_row = lntab_row.clone();
+            let mut row_cnt = 0;
+            let mut pc = 0;
```

Comment:
Isn't it the offset into `ops`? Given that `ops` represents operations, aka instructions, and we advance it to the next one after execution, the naming seems apt to me (though personally I wouldn't abbreviate everything everywhere).

Personally, I am not sure I follow how how "row offset" is more fitting: it's neither the offset of the current row nor an offset into the current row, is it?

Be that as it may, I suspect that once we use `ReadRaw` here this variable will be gone entirely, so perhaps this discussion is moot?

Reply:
ah, so we treat this "ops" as "a program" in itself, ok. Agree, it's still confusing. And yeah, row_off doesn't make sense, it's more of ops_off. But as you said, might be irrelevant.